### DISCUSSION TITLE
Project Status Update

### DISCUSSION THREAD

**Discussion by gabebott** [USU-ITLS-x265-23-summer/class-repo#181] [General]
This week I developed a game plan for attacking this memory game project. I concluded that the following tasks needed to be accomplished in the next few weeks and throughput this project. I think that this will be a challenging project for me, but I think that I will be able to accomplish it if I follow these steps and keep checking things off my list. 

- Create the basic HTML structure for the game board, cards, and any other necessary elements.
- Define CSS classes or IDs to style and manipulate the elements.
- Apply CSS styles to the game board, cards, and other elements to create an attractive and user-friendly interface.
- Use CSS animations or transitions to provide visual feedback for card flips and matches.
- Write JavaScript functions to handle game logic, such as flipping cards, checking for matches, and updating the game state.
- Implement a card shuffling algorithm to randomize the card positions at the start of each game.
- Add event listeners to handle user interactions, such as clicking on cards.
- Implement the logic to track the number of moves, the elapsed time, and the number of matched pairs.
- Display the game statistics, such as the number of moves and the elapsed time, to the user.
- Add a restart button to reset the game and start a new round.
- Check for the win condition when all card pairs have been matched.
- Display a congratulatory message or prompt the user to play again.

My goal for the upcoming week is to develop a high fidelity wireframe for the game. Coming from a UX background, a high fidelity wireframe helps me to visualize the design while also figuring out some of the small issues that could pop up in a prototype phase.

I also hope to decide on a theme for images to put on the cards in the memory game. I think that it should definitely be a consistent theme, so I could do animals, or fruit, or sports teams or something like that. I have to decide on that and then obviously implement the images into the project. As of right now that is my plan for this upcoming week and I feel like I am on track of where I want to be right now! If any of you have any experience working with games or projects similar to this please reach out and I would love to pick your brains.

**Comment by jakegros** [comment]
This sounds awesome.  I think you have a good plan and vision going.  I have definitely been wanting to create games with the skills we are learning and plan to even after this course.  However, thus far, I have only created them with block code.  Although I don't have experience creating them yet, feel free to bounce any questions or ideas off me if you want some feedback.
